src/main/kotlin/com/ampnet/projectservice/service/impl/OrganizationInviteServiceImpl.kt

```diff
@@ -35,36 +34,32 @@ class OrganizationInviteServiceImpl(
 
     companion object : KLogging()
 
-    private val adminRole: Role by lazy { roleRepository.getOne(OrganizationRoleType.ORG_ADMIN.id) }
-    private val memberRole: Role by lazy { roleRepository.getOne(OrganizationRoleType.ORG_MEMBER.id) }
-
     @Transactional
-    override fun sendInvitation(request: OrganizationInviteServiceRequest): OrganizationInvitation {
+    override fun sendInvitation(request: OrganizationInviteServiceRequest) {
         val invitedToOrganization = organizationService.findOrganizationById(request.organizationUuid)
             ?: throw ResourceNotFoundException(
                 ErrorCode.ORG_MISSING,
                 "Missing organization with id: ${request.organizationUuid}"
             )
-
-        inviteRepository.findByOrganizationUuidAndEmail(request.organizationUuid, request.email).ifPresent {
-            throw ResourceAlreadyExistsException(
-                ErrorCode.ORG_DUPLICATE_INVITE,
-                "User is already invited to join organization"
+        inviteRepository.findByOrganizationUuidAndEmailIn(request.organizationUuid, request.emails)
+            .whenNotNullNorEmpty { invitations ->
```

Comment:
Why check for Empty? Can OrganizationInvitation be empty?!

Reply:
The list can me empty i.e. size = 0 in case there is no invitation in the database matching the email.

Reply:
Ok, then I assume a list cannot be null? It is a weird case if the list can be null or empty.

Reply:
I will correct the extension function to check only when it is NotEmpty